core/src/main/java/hudson/console/ConsoleLogFilter.java

```diff
@@ -48,8 +55,41 @@
     /**
      * Called on the start of each build, giving extensions a chance to intercept
      * the data that is written to the log.
+     *
+     * @deprecated as of 1.630. Use {@link #decorateLogger(Run, OutputStream)}
      */
-    public abstract OutputStream decorateLogger(AbstractBuild build, OutputStream logger) throws IOException, InterruptedException;
+    public OutputStream decorateLogger(AbstractBuild build, OutputStream logger) throws IOException, InterruptedException {
+        if (Util.isOverridden(ConsoleLogFilter.class, getClass(), "decorateLogger", Run.class, OutputStream.class)) {
+            // old client calling newer implementation. forward the call.
+            return decorateLogger((Run) build, logger);
+        } else {
+            // happens only if the subtype fails to override neither decorateLogger method
+            throw new AssertionError("The plugin '" + this.getClass().getName() + "' still uses " +
+                    "deprecated decorateLogger(AbstractBuild,OutputStream) method. " +
+                    "Update the plugin to use setUp(Run,OutputStream) instead.");
+        }
+    }
+
+    /**
+     * Called on the start of each build, giving extensions a chance to intercept
+     * the data that is written to the log.
+     *
+     * <p>
+     * Even though this method is not marked 'abstract', this is the method that must be overridden
```

Comment:
There is not. It would be very useful in this and similar cases to have an `@Abstract` annotation with an associated processor which rejects class that do not override the method. @kohsuke POTD?